asm.s

```diff
@@ -1,156 +1,196 @@
+  .cfi_sections .debug_frame
+
   .section .text.__bkpt
   .global __bkpt
   .thumb_func
+  .cfi_startproc
 __bkpt:
   bkpt
   bx lr
+  .cfi_endproc
   .size __bkpt, . - __bkpt
 
   .section .text.__control_r
   .global __control_r
   .thumb_func
+  .cfi_startproc
 __control_r:
   mrs r0, CONTROL
   bx lr
+  .cfi_endproc
   .size __control_r, . - __control_r
 
   .section .text.__control_w
   .global __control_w
   .thumb_func
+  .cfi_startproc
 __control_w:
   msr CONTROL, r0
   bx lr
+  .cfi_endproc
   .size __control_w, . - __control_w
 
 
   .section .text.__cpsid
   .global __cpsid
   .thumb_func
+  .cfi_startproc
 __cpsid:
   cpsid i
   bx lr
+  .cfi_endproc
   .size __cpsid, . - __cpsid
 
   .section .text.__cpsie
   .global __cpsie
   .thumb_func
+  .cfi_startproc
 __cpsie:
   cpsie i
   bx lr
+  .cfi_endproc
   .size __cpsie, . - __cpsie
 
   .section .text.__delay
   .global __delay
   .syntax unified
   .thumb_func
+  .cfi_startproc
 __delay:
 1:
   nop
   subs r0, #1
   bne 1b  // Branch to 1 instead of __delay does not generate R_ARM_THM_JUMP8 relocation, which breaks linking on the thumbv6m-none-eabi target
   bx lr
+  .cfi_endproc
   .size __delay, . - __delay
 
   .section .text.__dmb
   .global __dmb
   .thumb_func
+  .cfi_startproc
 __dmb:
   dmb 0xF
   bx lr
+  .cfi_endproc
   .size __dmb, . - __dmb
 
   .section .text.__dsb
   .global __dsb
   .thumb_func
+  .cfi_startproc
 __dsb:
   dsb 0xF
   bx lr
+  .cfi_endproc
   .size __dsb, . - __dsb
 
   .section .text.__isb
   .global __isb
   .thumb_func
+  .cfi_startproc
 __isb:
   isb 0xF
   bx lr
+  .cfi_endproc
   .size __isb, . - __isb
 
   .section .text.__msp_r
   .global __msp_r
   .thumb_func
+  .cfi_startproc
 __msp_r:
   mrs r0, MSP
   bx lr
+  .cfi_endproc
   .size __msp_r, . - __msp_r
 
   .section .text.__msp_w
   .global __msp_w
   .thumb_func
+  .cfi_startproc
```

Comment:
this subroutine may modify the stack pointer depending on whether MSP or PSP is currently being used so it requires a (rather complex?) register rule so let's not add CFI to it in this PR